WordPress/Sniffs/Theme/NoHardCodedUrlsSniff.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * WordPress_Sniffs_Theme_NoHardCodedUrlsSniff.
+ *
+ * WARNING Verify that no hard-coded URLs are found. Manual verification necessary.
+ * An exception is made for the Author URI and (Theme) URI as set in the style.css header
+ * as well as links to wordpress.org. Links in the text of PHP/JS comments should be
+ * excluded from this check.
+ * Ref: https://make.wordpress.org/themes/handbook/review/required/theme-check-plugin/#info
+ * Theme check file covering this rule:
+ * https://github.com/Otto42/theme-check/blob/master/checks/links.php
+ *
+ * @category PHP
+ * @package  PHP_CodeSniffer
+ * @author  Kevin Archibald
+ */
+class WordPress_Sniffs_Theme_NoHardCodedUrlsSniff implements PHP_CodeSniffer_Sniff
+{
+  /**
+   * Returns an array of tokens this test wants to listen for.
+   *
+   * @return array
+   */
+  public function register()
+  {
+      return array(
+        T_STRING,
+        T_CONSTANT_ENCAPSED_STRING,
+        T_INLINE_HTML,
+      );
+  }// end register()
+
+  /**
+   * Processes this test, when one of its tokens is encountered.
+   *
+   * @param PHP_CodeSniffer_File $phpcsFile The file being scanned.
+   * @param int                  $stackPtr  The position of the current token
+   *                                        in the stack passed in $tokens.
+   *
+   * @return void
+   */
+  public function process(PHP_CodeSniffer_File $phpcsFile, $stackPtr)
+  {
+      $tokens = $phpcsFile->getTokens();
+      $token  = $tokens[$stackPtr];
+
+      /** If the token contains this string, exclude it
+       * ToDo - How can we account for an allowed footer link?
+       * If we require the footer link to be an Author URI or Theme URI it's possible
+       */
+      $allowed_strings = array(
+        'gmpg.org',
+        'wordpress.org',
+        'wordpress.com',
+        'schema.org',
+        'fonts.googleapis',
+      );
+      // regex borrowed from themecheck (borrowed from TAC), modified for url by Kevin
+      if ( preg_match_all( '#(http://|https://|ftp://|//)([[:alnum:]\-\.])+(\\.)([[:alnum:]]){2,4}([[:blank:][:alnum:]\/\+\=\%\&\_\\\.\~\?\-]*)#' , $token['content'], $matches, PREG_SET_ORDER )  &&
+          substr( trim( $token['content'] ), 0, 4 ) !== '<!--'  )
+      {
+        // allowed strings
+        foreach( $matches as $match )
+        {
+          $add_warning_flag = true;
+          foreach( $allowed_strings as $string ) {
+            if( strpos( $match[0] , $string ) )$add_warning_flag = false;
+          }
+          if( true == $add_warning_flag )
```

Comment:
Ok, I get it now. True that it's not needed, but I *would* recommend it as a best practice as it makes it clearer what is being tested. I'd also recommend making it a strict check ``===`. It can also prevent bugs if someone would change something in the code above (for instance turn it into a counter or something) or make those bugs more easily discoverable.
General rule of thumb: _always check for type (strict checks or explicit type check depending on the context) & what you want to know_.